viewer/lib/worldrenderer.js

```diff
@@ -60,23 +64,43 @@ class WorldRenderer {
     }
   }
 
-  setVersion (version) {
+  resetWorld () {
+    this.active = false
     for (const mesh of Object.values(this.sectionMeshs)) {
       this.scene.remove(mesh)
     }
     this.sectionMeshs = {}
+    for (const worker of this.workers) {
+      worker.postMessage({ type: 'reset' })
+    }
+  }
+
+  setVersion (version) {
+    this.version = version
+    this.resetWorld()
+    this.active = true
     for (const worker of this.workers) {
       worker.postMessage({ type: 'version', version })
     }
 
-    loadTexture(`textures/${version}.png`, texture => {
+    this.updateTexturesData()
+  }
+
+  updateTexturesData () {
+    loadTexture(this.texturesDataUrl || `textures/${this.version}.png`, texture => {
       texture.magFilter = THREE.NearestFilter
       texture.minFilter = THREE.NearestFilter
       texture.flipY = false
       this.material.map = texture
     })
 
-    loadJSON(`blocksStates/${version}.json`, blockStates => {
+    const loadBlockStates = () => {
```

Comment:
why is this needed? didn't it work before?

Reply:
I don't understand the phrasing of the question "didn't it work before?" - it's just an extension of the previous logic. Still works, but now you can override logic of getting block states via exposed property so you can skip network request. Ideally it should also set to this property if wasn't set before after request, fixed in my version but not here but that's not a big issue imo.